src/auth/jwtaccess.ts

```diff
@@ -65,12 +69,15 @@ export class JWTAccess {
    * @returns An object that includes the authorization header.
    */
   getRequestHeaders(url: string, additionalClaims?: Claims): Headers {
+    // Return cached authorization headers, unless we are within
+    // EXPIRY_DELTA ms of them expiring:
     const cachedToken = this.cache.get(url);
-    if (cachedToken) {
-      return cachedToken;
+    const now = Date.now();
+    if (cachedToken && cachedToken.expiration - now > EXPIRATION_DELTA) {
```

Comment:
question: why is this different from `oauth2client.isTokenExpiring` ? This uses the value `eagerRefreshThresholdMillis` and defaults to 300000 (5 minutes).

Reply:
No great reason, just thought 5 minutes is perhaps more than we need? perhaps I should be consistent.

Reply:
totally your call! Just a callout to make sure you were aware the logic is different

Reply:
we're now using `eagerRefreshThresholdMillis` for consistency.